client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts

```diff
@@ -20,31 +21,35 @@ import {
 } from '../InteractionsTab/useInteractionsTab';
 
 const useContactQuestioning = (parameters: useContactQuestioningParameters): useContactQuestioningOutcome => {
-    const {setAllContactedInteractions, allContactedInteractions, setFamilyRelationships, setContactStatuses} = parameters;
+    const {id, setAllContactedInteractions, allContactedInteractions, setFamilyRelationships, setContactStatuses} = parameters;
+    
     const userId = useSelector<StoreStateType, string>(state => state.user.id);
     const epidemiologyNumber = useSelector<StoreStateType, number>(state => state.investigation.epidemiologyNumber);
-    const { handleDuplicateIdsError } = useDuplicateContactId();
+    
+    const { checkDuplicateIds, handleDuplicateIdsError } = useDuplicateContactId();
 
-    const saveContact = (interactedContact: InteractedContact) => {
-        const contacts = [interactedContact];
-        const contactsSavingVariable = {
-            unSavedContacts: {contacts}
-        }
-        logger.info({
-            service: Service.CLIENT,
-            severity: Severity.LOW,
-            workflow: 'Saving single contact',
-            step: `launching server request with parameter: ${JSON.stringify(contactsSavingVariable)}`,
-            user: userId,
-            investigation: epidemiologyNumber
-        });
-        if(!checkForSpecificDuplicateIds(interactedContact.identificationNumber, interactedContact.id)) {
-            interactedContact.identificationNumber = '';
-            const changedInteractedContact = allContactedInteractions.findIndex(currContact => currContact.id === interactedContact.id);
-            allContactedInteractions.splice(changedInteractedContact, 1, interactedContact);
-            setAllContactedInteractions(allContactedInteractions);
-            handleDuplicateIdsError(interactedContact.identificationNumber, userId, epidemiologyNumber);
+    const saveContact = (interactedContact: InteractedContact) : boolean => {
+        const isDuplicateId = allContactedInteractions.some((contact: InteractedContact) => 
+            contact.id !== interactedContact.id &&
+            interactedContact.identificationNumber !== '' && 
+            interactedContact.identificationNumber !== null &&
```

Comment:
you can replace these with 
```suggestion
            !interactedContact.identificationNumber  &&
```

Reply:
@MayaSasson the genius

Reply:
I'll do it as you suggest below

Reply:
you can close it

Reply:
fixed